sherpa/astro/data.py

```diff
@@ -1761,27 +1761,87 @@ def get_specresp(self, filter=False):
 
         return newarf
 
-    # The energy bins can be grouped or ungrouped.  By default,
-    # they should be grouped if the data are grouped.  There are
-    # certain contexts (e.g., plotting) where we will retrieve the
-    # energy bins, and later filter the data; but filtering
-    # is automatically followed by grouping.  Grouping the data
-    # twice is an error.
     def _get_ebins(self, response_id=None, group=True):
+        """Return the low and high edges of the independent axis.
+
+        This method is badly named as it will return values in either
+        channel or energy units, depending on the units setting and
+        the associated response information. It also returns the
+        approximation of the mapping from channel space to energy -
+        that is the E_MIN and E_MAX columns from the RMF EBOUNDS block
+        rather than from the ENERG_LO and ENERG_HI columns from the
+        MATRIX block.
+
+        Parameters
+        ----------
+        response_id : int or None, optional
+            The response to use when units are not "channel". The
+            default is to use the default response identifier.
+        group : bool, optional
+            Should the current grouping setting be applied. This is
+            only used if the "grouped" attribute is set.
+
+        Returns
+        -------
+        lo, hi : ndarray
+            The low and high edges of each bin, in either channels or
+            keV: energy is used if the units setting is not energy and
```

Comment:
Is this really true: "energy is used if the units setting is not energy"? Maybe there is a "not" too many?

Reply:
Or it shoud say "is not channel" instead of "is not energy"?

Reply:
Hopefully the re-worked version is better.